internal/route/request_executor.go

```diff
@@ -116,22 +127,22 @@ func (r *RequestExecutor) useCache(requestBody jsonrpc.RequestBody) bool {
 	return r.cache.ShouldCacheMethod(requestBody.GetMethod())
 }
 
-func (r *RequestExecutor) retrieveOrCacheRequest(httpReq *http.Request, requestBody jsonrpc.SingleRequestBody, configToRoute *config.UpstreamConfig) (jsonrpc.ResponseBody, *http.Response, bool, error) {
+func (r *RequestExecutor) retrieveOrCacheRequest(httpReq *http.Request, requestBody jsonrpc.SingleRequestBody, configToRoute *config.UpstreamConfig) (jsonrpc.ResponseBody, *HttpResponse, bool, error) {
 	var (
-		respBody jsonrpc.ResponseBody
-		resp     *http.Response
+		jsonRPCRespBody jsonrpc.ResponseBody
+		httpResp        *HttpResponse
 	)
 
 	originFunc := func() (*jsonrpc.SingleResponseBody, error) {
 		var err error
 
 		// Any errors will result in respBody and resp being nil.
-		respBody, resp, err = r.getResponseBody(httpReq, &requestBody, configToRoute) //nolint:bodyclose // linter bug
+		jsonRPCRespBody, httpResp, err = r.getResponseBody(httpReq, &requestBody, configToRoute) //nolint:bodyclose // linter bug
```

Comment:
A lil bit of renaming to jsonRPCRespBody and httpResp in a few places. Thought this naming would be more clear.